core/src/main/scala/cromwell/core/callcaching/docker/DockerHashActor.scala

```diff
@@ -0,0 +1,209 @@
+    import GraphDSL.Implicits._
+    
+    def mapContextToFlow(dockerHashContext: DockerHashContext) = {
+      dockerRegistryFlows.indexWhere(_.accepts(dockerHashContext.dockerImageID)) match {
+          // requests with no matching host are sent to the last port
+        case -1 => dockerRegistryFlows.length
+        case found => found
+      }
+    }
+
+    val partitionFlows = builder.add(Partition[DockerHashContext](dockerRegistryFlows.size + 1, mapContextToFlow))
+    val mergeFlows = builder.add(Merge[(DockerHashResponse, DockerHashContext)](dockerRegistryFlows.size + 1))
+    
+    // Build a partial flow for each dockerRegistryFlow available
+    val flows = dockerRegistryFlows map { _.buildFlow() }
+
+    // Connect each flow to one output port of the broadcast, and merge them all back together
+    dockerRegistryFlows.indices foreach { i =>
+      partitionFlows.out(i) ~> flows(i) ~> mergeFlows.in(i)
+    }
+
+    val noMatch = partitionFlows.out(dockerRegistryFlows.size).map(dockerContext => (DockerHashUnknownRegistry(dockerContext.dockerImageID), dockerContext)).outlet
+    
+    noMatch ~> mergeFlows.in(dockerRegistryFlows.size)
+    
+    FlowShape(partitionFlows.in, mergeFlows.out)
+  }
+
+  private val queue = Source.queue[DockerHashContext](queueBufferSize, OverflowStrategy.dropNew)
+    .via(dockerFlow)
+    .alsoTo(updateCacheSink)
+    .to(replySink)
+    .withAttributes(ActorAttributes.supervisionStrategy(decider))
+    .run()
+  
+  override def receive: Receive = {
+    // When receiving a request, just enqueue it and pipe the result back to self
+    // Note: could be worth having another actor to send it to, to prevent overflowing this mailbox
+    case request: DockerHashRequest =>
+      val replyTo = sender()
+      
+      checkCache(request.dockerImageID) match {
+        case Some(cacheHit) => replyTo ! cacheHit
+        case None => sendToStream(DockerHashContext(request, replyTo))
+      }
+    case EnqueueResponse(Enqueued, dockerHashContext) => // All good !
+    case EnqueueResponse(Dropped, dockerHashContext) => dockerHashContext.replyTo ! DockerHashBackPressure(dockerHashContext.request)
+      // In any of the below case, the stream is in a state where it will not be able to receive new elements.
+      // This means something blew off so we can just restart the actor to re-instantiate a new stream
+    case EnqueueResponse(QueueClosed, dockerHashContext) =>
+      val exception = new RuntimeException(s"Failed to enqueue docker hash request ${dockerHashContext.request}. Queue was closed")
+      logAndRestart(exception)
+    case EnqueueResponse(QueueOfferResult.Failure(failure), dockerHashContext) => 
+      logAndRestart(failure)
+    case FailedToEnqueue(throwable, dockerHashContext) =>
+      logAndRestart(throwable)
+  }
+  
+  private def logAndRestart(throwable: Throwable) = {
+    log.error("Failed to process docker hash request", throwable)
```

Comment:
Would `warning` be better ? This is really not supposed to happen so if it does we should be able to see it I think